src/error/fowler-noise.lisp

```diff
@@ -0,0 +1,200 @@
+  (make-instance 'fowler-pure-state-qvm
+                 :noise-class (fowler-qvm-noise-class qvm)
+                 :noise-probability (fowler-qvm-noise-probability qvm)
+                 :number-of-qubits (number-of-qubits qvm)
+                 :state (make-instance 'pure-state
+                                       :num-qubits (number-of-qubits qvm)
+                                       :amplitudes (alexandria:copy-array
+                                                    (qvm::amplitudes (qvm::state qvm))))
+                 :gate-definitions (qvm::gate-definitions qvm)
+                 :wait-function (qvm::wait-function qvm)
+                 :classical-memory-subsystem (qvm::classical-memory-subsystem qvm)))
+
+(defmethod copy-fowler-qvm ((qvm fowler-stabilizer-qvm))
+  "Makes a \"deep copy\" of (the quantum part of) `QVM', of type `FOWLER-STABILIZER-QVM'.
+
+WARNING: Deep copy does _not_ include the classical memory subsystem. Writes to the returned QVM's classical memory will be visible to the original QVM's classical memory, and vice versa."
+  (make-instance 'fowler-stabilizer-qvm
+                 :noise-class (fowler-qvm-noise-class qvm)
+                 :noise-probability (fowler-qvm-noise-probability qvm)
+                 :tableau (alexandria:copy-array (qvm::stabilizer-qvm-tableau qvm))
+                 :gate-definitions (qvm::gate-definitions qvm)
+                 :wait-function (qvm::wait-function qvm)
+                 :classical-memory-subsystem (qvm::classical-memory-subsystem qvm)))
+
+(defmethod compile-loaded-program ((qvm fowler-qvm))
+  qvm)
+
+(define-condition unknown-depolarizing-effect (error)
+   ((instruction :reader unknown-depolarizing-effect-instruction :initarg :instruction))
+   (:report (lambda (condition stream)
+              (format stream "FOWLER-QVM doesn't know how to follow ~a with depolarizing noise" 
+                      (unknown-depolarizing-effect-instruction condition)))))
+
+;;;
+;;; read individual flags out of the error-type bitvector
+;;;
+
+(declaim (inline fowler-qvm-noise-identity?
+                 fowler-qvm-noise-1Q-gate?
+                 fowler-qvm-noise-2Q-gate?
+                 fowler-qvm-noise-reset?
+                 fowler-qvm-noise-readout?))
+
+(defun fowler-qvm-noise-identity? (qvm)
+  (logbitp 0 (fowler-qvm-noise-class qvm)))
+
+(defun fowler-qvm-noise-1Q-gate? (qvm)
+  (logbitp 1 (fowler-qvm-noise-class qvm)))
+
+(defun fowler-qvm-noise-2Q-gate? (qvm)
+  (logbitp 2 (fowler-qvm-noise-class qvm)))
+
+(defun fowler-qvm-noise-reset? (qvm)
+  (logbitp 3 (fowler-qvm-noise-class qvm)))
+
+(defun fowler-qvm-noise-readout? (qvm)
+  (logbitp 4 (fowler-qvm-noise-class qvm)))
+
+;;;
+;;; basic qvm behavior definitions
+;;;
+
+(defmethod %fast-apply ((qvm fowler-pure-state-qvm) instr-name qubit)
```

Comment:
this method doesn't actually exist in the QVM (maybe it used to; I don't remember)

because of this, it might be a good idea to `defgeneric` it somewhere, or rename it something more to your liking since it's private to this package

Reply:
Oh. We definitely didn't come up with the name, so it must have gotten cleaned up in the past couple years.

I'll add a `defgeneric`, but I won't rename it. That's more headache than it's worth.